File: algorithms/leetcode/easy/1672-richest-customer-wealth/kotlin/Solution.kt
// 1672. Richest Customer Wealth
// 解題日期: 13天前

class Solution {
    fun maximumWealth(accounts: Array<IntArray>): Int {
        var richest = 0
        for (customer in accounts) {
            var current = 0
            for (wealth in customer) {
                current += wealth
                if (current > richest) {
                    richest = current
                }
            }
        }
        return richest
    }
}

File: algorithms/leetcode/easy/1672-richest-customer-wealth/solution.py
from typing import List

class Solution:
    def maximumWealth(self, accounts: List[List[int]]) -> int:
        """
        找出最富有的客戶財富總和

        新解法：使用生成器表達式和 max 函數
        更簡潔，一行解決
        """
        return max(sum(customer) for customer in accounts)
